pkgs/jupiter-hw-support/bios-fwupd.nix

```diff
@@ -0,0 +1,71 @@
+# To install the BIOS update:
+#
+# 1. Disable Secure Boot, or have a `fwupd-efi` with `fwupdx64.efi.signed`.
+# 2. Set `OnlyTrusted=false` in `/etc/fwupd/daemon.conf`.
+#    (WARNING: This may make your device less secure!)
+# 3. Run `sudo fwupdmgr install /path/to/bios.cab`
+
+{ lib
+, runCommand
+, callPackage
+, jupiter-hw-support
+, gcab
+
+# The following parameters are for your enjoyment of flashing custom/older BIOSes:
+
+# The BIOS file (e.g., "F7A0110_sign.fd") or a directory containing it.
+, biosFile ? jupiter-hw-support.src
+
+# The BIOS version (e.g., "0110").
+# If null, parsed from the BIOS file name.
+, biosVersion ? null
+
+# The release date of the BIOS for the fwupd manifest.
+, releaseDate ? "1970-01-01"
+
+# The human-readable version for the package.
+, packageVersion ? jupiter-hw-support.src.version
+}:
+
+let
+  version = if packageVersion != null then packageVersion else jupiter-hw-support.version;
+
+in runCommand "steamdeck-bios-fwupd-${version}" {
+  inherit biosFile biosVersion releaseDate;
+
+  nativeBuildInputs = [ gcab ];
+
+  meta = with lib; {
+    description = "Steam Deck BIOS for fwupd";
+    license = licenses.unfreeRedistributableFirmware;
+  };
+} ''
+  if [[ -d "$biosFile" ]]; then
+    biosFile=$(find "$biosFile" -regextype posix-extended -type f -regex ".*/F7A.*_sign.fd" | head -1)
+  fi
+
+  >&2 echo ":: Found BIOS file $biosFile"
+
+  if [[ -z "$biosVersion" ]]; then
+    versionPair=$(basename "$biosFile" | sed 's/^.*F7A\([0-9]\{2\}\)\([0-9]\{2\}\).*$/\1 \2/; t; q1')
+  else
+    versionPair=$(echo "$biosVersion" | sed 's/^\([0-9]\{2\}\)\([0-9]\{2\}\)$/\1 \2/; t; q1')
+  fi
+
+  get_numeric_ver() { printf "%d" "0x00$100$2"; }
```

Comment:
Some observations, a lot of conjecture:

 - the version field is (unsourced) an at least 32 bit wide integer.
 - the bios vendor apparently used two "wide" chars to save...
 - ... an hexadecimal-encoded identifier of the decimal digits

So in e.g. `F7A0110` we have the two integers `0x01` and `0x10` packed into two UTF-16 chars, which would default to big-endian. `0x00__00__ → 0x00010010 → 65552`.

At least this is the most plausible scenario I could muster for the way it's being encoded.